src/babashka/nrepl/server/middleware.clj

```diff
@@ -144,3 +144,21 @@
          (disj #'wrap-process-message)
          (conj op-handler)))))
 
+(defn
+  middleware->transducer
+  "Return a transducer from a `middleware`.
+  Preserves `::requires` and `::expects` metadata of `middlware`.
+  See https://github.com/babashka/babashka.nrepl/blob/master/doc/middleware.md."
+  ([middleware]
+  (let [f
+        (fn
+          [rf]
+          (fn
+            ([] (rf))
+            ([result] (rf result))
+            ([result input] ((middleware #(rf result %)) input))))
+        {::keys [requires expects]} (meta middleware)]
+    (with-meta f
+      (cond-> {}
+        requires (assoc ::requires (into #{} (map requiring-resolve) requires))
```

Comment:
Should requiring-resolve be evaluated in SCI here?

Reply:
The metadata references `babashka.nrepl.server.middleware/wrap-read-msg` etc. So those are jvm symbols; 

But you just made me realize that there is no mechnism right now for the user to sort user land middleware. 


1 idea is to split the jvm `:require`'d middleware symbols from the user land symbols by using another key. Like 
`babashka.nrepl.server.middleware/expects`, that means jvm symbols and 
`:bb-user-middleware/expects` that means user middleware. 

To make `middleware->graph` work, we could add a middleware-name metadata on the anonymous function that `middleware->transducer` returns.

Reply:
Using `requiring-resolve` here isn't great since this will blow up native image size. Just compile bb with these changes and you will see that it will go over 110mb (from 75mb).
We should only do this within SCI (if it's at all necessary, which should be demonstrated by an accompanying test). If it's necessary we can expose `babashka.nrepl.server.middleware/wrap-read-msg` etc via a SCI namespace.

> cider-nrepl doesn't seem to depend on it

nrepl also sorts, see [here](https://github.com/nrepl/nrepl/blob/4715a016eb8e3046d7d4466a02663c61e53a5614/src/clojure/nrepl/middleware.clj#L96-L184).

Note that babashka nrepl already has similar sorting.

Reply:
We already needed the sorting for the current tests. 

Ok I think we want to expose `babashka.nrepl.server.middleware/wrap-read-msg` etc via a SCI namespace, then. 
How do I define this namespace and expose it? 
Do I make an ns with sci/create-ns in a nrepl source file like sci_middleware.clj. 

Do we then add the namespace in babashka `main` ? 

I see nrepls sorting now - I think we then want to support the same sort (cider-nrepl also uses it extensively I see now).
It means we support `:requires`, `:expects` and `:handles`. 

Not sure yet if we should modify `middleware->graph` or add a second sort for sci-middleware... I am thinking about copying the nrepl one.